Strassen Matrix Multiplication/Matrix.swift

```diff
@@ -0,0 +1,246 @@
+//
+//  Matrix.swift
+//  
+//
+//  Created by Richard Ash on 10/28/16.
+//
+//
+
+import Foundation
+
+enum RowOrColumn {
```

Comment:
Opps, you're right. I believe this was a discussion point as part of the generics manifesto that's planned for Swift 4.0.

I don't think a workaround is possible at the moment.